friends/src/components/FriendPage.js

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { fetchFriend, editFriend } from '../actions';
+
+class FriendPage extends Component {
+  state = {
+    name: '',
+    age: '',
+    email: ''
+  };
+
+  submitHandler = event => {
+    event.preventDefault();
+    this.props.editFriend({ id: this.props.friend.id, ...this.state });
+    this.setState({
+      name: '',
+      age: '',
+      email: ''
+    });
+  };
+
+  changeHandler = event => {
+    this.setState({ [event.target.name]: event.target.value });
+  };
+
+  componentDidMount() {
+    this.props.fetchFriend(this.props.match.params.id);
+    this.setState({
+      name: this.state.name,
+      age: this.state.age,
+      email: this.state.email
```

Comment:
There are a couple of reasons this isn't working. The first is that you're trying to set state with this.state.name -- I think you meant to use props here. The second reason it's not working is because it's async. Your flow is like this: friend is fetched --> through redux, friend on redux state gets updated --> updated info is sent as a prop since you've mappedStateToProps. HOWEVER(!) this setState is getting executed before all of that has been completed. In other words, you're just setting everything to undefined here.

TWO solutions: 
1.) most simple and straightforward -- just do your axios call right here within componentDidMount, and then in a .then statement, set the state using the response.
2.) Keeping your existing flow, you can add in a componentDidUpdate lifecycle method. That will basically listen for changes to props. Once a change occurs, then set the state with the new props. Here's what that would look like (tested and working):
```
  componentDidUpdate(prevProps) {
    if (this.props.friend.name !== prevProps.friend.name) {
      this.setState({
        name: this.props.friend.name,
        age: this.props.friend.age,
        email: this.props.friend.email
      })
    }
  }
```